hazelcast/src/main/java/com/hazelcast/transaction/impl/TransactionLog.java

```diff
@@ -38,53 +37,40 @@
  * the TransactionLog will contains 4 {@link TransactionLogRecord} instances.
  *
  * planned optimization:
- * Most transaction will be small, but an linkedlist + hashmap is created. Instead use an array and do a
+ * Most transaction will be small, but an linkedhashmap is created. Instead use an array and do a
  * linear search in that array. When there are too many items added, then enable the hashmap.
```

Comment:
Out of curiosity I've implemented the `TransactionLog` using a plain `ArrayList` and linear search. [My benchmark|https://gist.github.com/gdela/48c235ccdd083944ae0e258547a81c1c] showed that actually there is no performance benefits in such implementation, even for small transactions:

`LinkedHashMap` based implementation:
```
MyTransactionBenchmark._10_baseline     avgt    6     ,317 ±  0,007  ms/op
MyTransactionBenchmark._10_backward     avgt    6     ,339 ±  0,004  ms/op
MyTransactionBenchmark._10_forward      avgt    6     ,344 ±  0,025  ms/op
MyTransactionBenchmark._10_random       avgt    6     ,332 ±  0,002  ms/op
```

`ArrayList` based implementation:
```
MyTransactionBenchmark._10_baseline    avgt    6    ,315 ± 0,007  ms/op
MyTransactionBenchmark._10_backward    avgt    6    ,341 ± 0,009  ms/op
MyTransactionBenchmark._10_forward     avgt    6    ,328 ± 0,006  ms/op
MyTransactionBenchmark._10_random      avgt    6    ,336 ± 0,005  ms/op
```

Though maybe my `ArrayList` based implementation was too naive.